src/app/dashboard-widgets/environment-widget/environment-widget.component.ts

```diff
@@ -1,232 +1,46 @@
+  DeploymentsService,
+  Space
+} from '../../../app/space/create/deployments/services/deployments.service';
 
 @Component({
   encapsulation: ViewEncapsulation.None,
   selector: 'fabric8-environment-widget',
   templateUrl: './environment-widget.component.html',
-  styleUrls: ['./environment-widget.component.less']
+  styleUrls: ['./environment-widget.component.less'],
+  providers: [DeploymentsService]
 })
-export class EnvironmentWidgetComponent extends AbstractWatchComponent  implements OnInit, OnDestroy {
+export class EnvironmentWidgetComponent implements OnInit {
 
-  loading: Subject<boolean> = new BehaviorSubject(true);
+  spaceId: Observable<string>;
+  appInfos: Observable<ApplicationAttributesOverview[]>;
   contextPath: Observable<string>;
-  appsCount: number = 0;
-  contextSubscription: Subscription;
-  currentContext: Context;
-  space: ConnectableObservable<Space>;
 
-  protected environmentCache: Map<string, EnvironmentDeployments> = new Map<string, EnvironmentDeployments>();
-  protected subscriberCache: Map<string, Subscription> = new Map<string, Subscription>();
-  protected appsSubject: Subject<AppDeployments[]> = new BehaviorSubject([]);
-  private listCache: Map<string, Observable<any[]>> = new Map<string, Observable<any[]>>();
-  newSubscription: Subscription;
+  private subscriptions: Subscription[] = [];
 
   constructor(private context: Contexts,
-              private spaceStore: SpaceStore,
-              private serviceService: ServiceService,
-              private routeService: RouteService,
-              public route: ActivatedRoute,
-              private deploymentConfigService: DeploymentConfigService,
-              private deploymentService: DeploymentService) {
-    super();
+              private spaces: Spaces,
+              private deploymentsService: DeploymentsService) {
+    this.spaceId = this.spaces.current.first().map(space => space.id);
   }
 
   ngOnInit() {
-    this.contextSubscription = this.context.current.subscribe((context) => {
-      this.currentContext = context;
-      let pathRegex = new RegExp(/^\/(.+?)\//);
-      let pathSections = pathRegex.exec(context.path);
-      if (pathSections && pathSections.length > 0) {
-        let id = pathSections[1];
-        this.spaceStore.load(id);
-
-        this.environmentCache.clear();
-        this.subscriberCache.clear();
-        this.listCache.clear();
-      }
-    });
-
-    this.contextPath = this.context.current.map(context => context.path);
-
-    this.space = this.spaceStore.resource
-      .distinctUntilChanged()
-      .debounce(space => ((space && space.environments) ? Observable.interval(0) : Observable.interval(1000)))
-      .do(space => {
-        if (space === null) {
-          this.appsSubject.next([]);
-          this.appsCount = 0;
-        }
+    this.subscriptions.push(
+      this.spaceId.subscribe((spaceId: string) => {
+        this.appInfos = this.deploymentsService.getAppAndEnvironments(spaceId);
       })
```

Comment:
`subscriptions` array is not being used for anything meaningful anymore, please remove